### DISCUSSION TITLE
current status of thingino?

### DISCUSSION THREAD

**Discussion by virmaior** [gtxaspec/wz_mini_hacks#765] [Q&A] [answered]
Thingino seems like it's the way to go moving forward from here.

Several questions but nowhere to ask them there.
1. is there a way to test it similar to wz_mini_hacks by putting into an SD card and using the hook
2. does it save to the SD card in similar minute increments?
3. does it support Ethernet over USB?
4. what still needs to be done to bring up to functionality?

**Comment by gtxaspec** [comment, marked as answer]
1. **Complete OS Replacement**: Thingino fully wipes the existing setup, effectively replacing the entire operating system, no current way to load it via sd card.  If you have physical access to the device, it's possible to backup the existing flash via USB, and flash thingino via USB as well using the Ingenic cloner Windows/Linux based utility.

2. **Recording with openRTSP / ffmpeg**: You can use openRTSP or ffmpeg for recording purposes, including down to the minute clips.  I haven't recorded to an SD card recently and need to run some additional tests. I'll provide an update once I have more information.

3. **Connectivity Options**: Yes, the system supports usb over ethernet, and USB direct connectivity as well.

4. **Ongoing Developments**:
   - **Audio in RTSP Stream**: We are currently working on incorporating audio support into the RTSP stream, which is presently video-only.
   - **Substream Support**: Development is also focused on adding substream capabilities, as it currently supports only the main stream via RTSP.

5. **Open Source Contributions**:
   - **Open Source Project**: All sources are completely open, from the build system, os environment, video streamer, audio streamer, etc.  If anyone has expertise in RTSP, C, or C++ programming, we encourage you to contribute!
   - **Build System Enhancements**: We've been diligently developing a user-friendly, modular, and extendable build system based on Buildroot. This includes automatic builds for devices using GitHub Actions CI, and testing across a broad array of Ingenic devices such as T10, T20, T21, T23, T30, and T31. Our aim is to support every Ingenic SOC model. Currently, the streamer is functional on all the SOCs mentioned, though some streamer code cleanup and device-specific fixes are still needed.

Regarding Wyze support, Wyze Cam V2, Pan V1, Pan V2, Cam V3, and the Pan V3 ( with a swap to an unlocked soc chip ), as well as any similar OEM devices such as the Atom, Hualai, and Personal Cam branded devices, are supported.

**Reply by virmaior** [reply]
very exciting work!
